tests/rptest/archival/s3_client.py

```diff
@@ -0,0 +1,181 @@
+                raise SlowDown()
+            else:
+                raise
+
+    @retry_on_slowdown()
+    def _get_object(self, bucket, key):
+        """Get object from S3"""
+        try:
+            return self._cli.get_object(Bucket=bucket, Key=key)
+        except ClientError as err:
+            self.logger.debug(f"error response {err}")
+            if err.response['Error']['Code'] == 'SlowDown':
+                raise SlowDown()
+            else:
+                raise
+
+    def get_object_data(self, bucket, key):
+        resp = self._get_object(bucket, key)
+        return resp['Body'].read()
+
+    def get_object_meta(self, bucket, key):
+        """Get object metadata without downloading it"""
+        resp = self._get_object(bucket, key)
+        # Note: ETag field contains md5 hash enclosed in double quotes that have to be removed
+        return S3ObjectMetadata(Bucket=bucket,
+                                Key=key,
+                                ETag=resp['ETag'][1:-1],
+                                ContentLength=resp['ContentLength'])
+
+    def write_object_to_file(self, bucket, key, dest_path):
+        """Get object and write it to file"""
+        resp = self._get_object(bucket, key)
+        with open(dest_path, 'wb') as f:
+            body = resp['Body']
+            for chunk in body.iter_chunks(chunk_size=0x1000):
+                f.write(chunk)
+
+    @retry_on_slowdown()
+    def _list_objects(self, bucket, token=None, limit=1000):
+        try:
+            if token is not None:
+                return self._cli.list_objects_v2(Bucket=bucket,
+                                                 MaxKeys=limit,
+                                                 ContinuationToken=token)
+            else:
+                return self._cli.list_objects_v2(Bucket=bucket, MaxKeys=limit)
+        except ClientError as err:
+            self.logger.debug(f"error response {err}")
+            if err.response['Error']['Code'] == 'SlowDown':
+                raise SlowDown()
+            else:
+                raise
+
+    def list_objects(self, bucket):
+        token = None
+        truncated = True
+        while truncated:
+            res = self._list_objects(bucket, token,
+                                     limit=4)  # TODO: limit by larger value
+            token = res.get('NextContinuationToken')
+            truncated = bool(res['IsTruncated'])
+            if 'Contents' in res:
+                for item in res['Contents']:
+                    yield S3ObjectMetadata(Bucket=bucket,
+                                           Key=item['Key'],
+                                           ETag=item['ETag'][1:-1],
+                                           ContentLength=item['Size'])
+
+
+if __name__ == '__main__':
+    import logging
+    cli = S3Client(region='myregion',
+                   access_key=u"AKIAIOSFODNN7EXAMPLE",
+                   secret_key=u"wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY",
```

Comment:
are these keys we want to be in the tree? and if not, should we change them now? :)